### PR TITLE
Create tests for the escrow smart contract

### LINKED ISSUE #7: Add a comprehensive testsuite
We have currently one test that in a way tests the whole flow. We should decouple the tests into few phases dictated by the different statuses the contract be in.

### PR DESCRIPTION
**Pull request by jjanii**
Closes #7

### CODE REVIEW

**Comment by brdlyptrs** [comment]
Looks good to me, dictation and code makes sense.

**Comment by rbval** [comment]
@gaieges they actually are in separate containers: python one in `api` and the node one in `contracts` container.

**Comment by gaieges** [comment]
@rbval sorry meant different images

File: contracts/Escrow.sol [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    uint256 private recordingOracleStake;
+
+    address private canceler;
+    address private eip20;
+
+    string private manifestUrl;
+    string private manifestHash;
+
+    string private intermediateManifestUrl;
+    string private intermediateManifestHash;
```
Comment (e271828-) [inline contracts/Escrow.sol]: Shouldn't these be intermediateResultsUrl and intermediateResultsHash?

Comment (jjanii) [reply contracts/Escrow.sol]: The contract itself is just a copypaste of the old `hmt.sol` file. I can change these to the ones you suggested if that's what they are supposed to be.

Comment (posix4e) [reply contracts/Escrow.sol]: No let's keep to the copy/paste approach

**Pull request merged**